app/models/concerns/fog_extensions/google/server.rb

```diff
@@ -71,11 +74,24 @@ def vm_ip_address
       end
 
       def associate_external_ip
-        public_ip_address.present?
+        external_nat_present?
       end
 
       private
 
+      def external_nat_present?
```

Comment:
we can probably simplify the method like:
```ruby
return false if network_interfaces.blank?
!!network_interfaces.detect do |nic|
  !nic[:access_configs].blank? && nic[:access_config].detect { |c| c[:name].eql?(EXTERNAL_NAT_NAME) }
end
```
It would feel bit cleaner not to have overly complex method to get a flag.

Reply:
Done!